src/interval.jl

```diff
@@ -469,6 +714,100 @@ function Base.merge(a::AbstractInterval, b::AbstractInterval)
     return Interval(left, right)
 end
 
+Base.union(x::AbstractInterval) = x
+
+const AbstractIntervals = Union{AbstractInterval, AbstractVector{<:AbstractInterval}}
+mergeclean(x, y) = unbunch(union(x), union(y))
+function Base.intersect(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx && iny, mergeclean(x, y)...)
+end
+function Base.union(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx || iny, mergeclean(x, y)...)
+end
+function Base.setdiff(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx && !iny, mergeclean(x, y)...)
+end
+function Base.symdiff(x::AbstractIntervals, y::AbstractIntervals)
+    return mergesets((inx, iny) -> inx ⊻ iny, mergeclean(x, y)...)
+end
+function Base.issubset(x::AbstractIntervals, y::AbstractIntervals)
+    return isempty(setdiff(x, y))
+end
+# may or may not be from Base (see top of `Intervals.jl`)
+function isdisjoint(x::AbstractIntervals, y::AbstractIntervals)
+    return isempty(intersect(x, y))
+end
+Base.in(x, y::AbstractVector{<:AbstractInterval}) = any(yᵢ -> x ∈ yᵢ, y)
+function Base.issetequal(x::AbstractIntervals, y::AbstractIntervals)
+    x, y = mergeclean(x,y)
+    return x == y || (all(isempty, bunch(x)) && all(isempty, bunch(y)))
+end
+
+Base.length(x::AbstractInterval) = 1
+
+# sort endpoints so that a closed left endpoint comes before
+# a closed right endpoint (used by `find_intersections`)
+struct EndpointOffset{E <: Endpoint}
+    data::E
+end
+offset(x::Endpoint) = EndpointOffset(x)
+function Base.isless(x::EndpointOffset, y::EndpointOffset)
+    if isequal(x.data, y.data)
+        return isless(offset_value(x.data), offset_value(y.data))
+    else
+        return isless(x.data, y.data)
+    end
+end
```

Comment:
Code is far more complex then it needs to be. What should be done is this can be turned into a custom `lt` function for `sort` and can be turned into:
```suggestion
function offset_lt(x::Endpoint, y::Endpoint)
   return (
       a.endpoint == b.endpoint && isleft(a) && isclosed(a) && !isleft(b) && isclosed(b) ||
       isless(a, b)
   )
end
```
This code is far easier to understand and is inline with the comment above. To do this however requires `unbunch` to take a `sortlt` keyword or another iteration where sorting is extracted from `unbunch`ing